src/slack.rs

```diff
@@ -68,16 +69,20 @@ impl Service for SlackService {
         let result = json::decode::<SlackUserListResponse>(&body).unwrap();
         assert!(result.ok);
         let users = result.members.iter().filter(|user| {
-            match (user.deleted, user.has_2fa) {
-                (true, _) => false,
-                (false, None) => true,
-                (false, Some(true)) => false,
-                (false, Some(false)) => true,
+            match (user.deleted, user.is_bot, user.has_2fa) {
+                (true, _, _) => false,
+                (false, true, _) => false,
+                (false, false, None) => true,
+                (false, false, Some(true)) => false,
+                (false, false, Some(false)) => true,
             }
         }).map(|user|
             User{
                 name: user.name.to_string(),
-                email: Some(user.profile.email.to_string()),
+                email: match user.profile.email {
+                    Some(ref m) => Some(m.to_string()),
```

Comment:
If the answer to this is "no", as I suspect it is, I believe you can just write `Some(user.profile.email.unwrap().to_string())`.